configure.ac

```diff
@@ -533,6 +533,10 @@ AC_SUBST([CRYPTO_LIBS])
 AC_SUBST([STRINGPARAM_HMAC])
 AM_CONDITIONAL([COND_USE_OPENSSL], [test "x$use_openssl" = "xyes"])
 
+AC_CHECK_FUNCS([setreuid setregid],,
+  AC_CHECK_FUNCS([setreuid32 setregid32],
+    AC_DEFINE([HAVE_LINUX_32BIT_SYSCALLS], 1, [32 bit system calls provided])))
+
```

Comment:
Imagine that setreuid() and setregid() are provided by libc but implemented using appropriate *32 syscalls,
this is what glibc does on old 32-bit architectures.
I don't think the check for functions provided by libc is a reliable way to decide which syscalls to invoke.
Wouldn't simple ifdef's for SYS_setresuid32/SYS_setreuid32/SYS_setregid32 work better?